### ISSUE TITLE
WS in quoted-pair (Trac #64)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#64]
I think quoted-pair is broken too. Merging your fix into RFC2616 gives:

```text
quoted-string  = ( <"> *(qdtext | quoted-pair ) <"> )
qdtext         = <any TEXT excluding '"' and '\'>
quoted-pair    = "\" CHAR
CHAR           = <any US-ASCII character (octets 0 - 127)>
]]>
```

                            
but that means you can do this:

```text
HTTP/1.1 200 OK
Warning: "Don't misparse \
this: it's really a single header!"
```

                
(if the receiving implementation follows the recommendations in 19.3 you
need to escape the LF instead of the CR, but it's otherwise the same.)

RFC 2822 updates RFC 822's quoted-pair rule to disallow CR, LF, and NUL.
We should probably make the same change.

Migrated from https://trac.ietf.org/ticket/64
```json
{
    "status": "closed",
    "changetime": "2008-04-03T13:29:45",
    "_ts": "1207229385000000",
    "description": "I think quoted-pair is broken too. Merging your fix into RFC2616 gives:\n\n\n{{{\nquoted-string  = ( <\"> *(qdtext | quoted-pair ) <\"> )\nqdtext         = <any TEXT excluding '\"' and '\\'>\nquoted-pair    = \"\\\" CHAR\nCHAR           = <any US-ASCII character (octets 0 - 127)>\n]]>\n}}}\n\n                            \nbut that means you can do this:\n\n\n{{{\nHTTP/1.1 200 OK\nWarning: \"Don't misparse \\\nthis: it's really a single header!\"\n}}}\n\n                \n(if the receiving implementation follows the recommendations in 19.3 you\nneed to escape the LF instead of the CR, but it's otherwise the same.)\n\nRFC 2822 updates RFC 822's quoted-pair rule to disallow CR, LF, and NUL.\nWe should probably make the same change.",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2007-12-22T02:48:07",
    "component": "p1-messaging",
    "summary": "WS in quoted-pair",
    "priority": "",
    "keywords": "",
    "milestone": "unassigned",
    "owner": "",
    "type": "editorial",
    "severity": ""
}
```

**Comment by mnot** [comment]
Trac update at 20080104T06:05:50: 
* mnot@pobox.com changed component from "" to "messaging"
* mnot@pobox.com changed milestone from "" to "unassigned"
* mnot@pobox.com changed owner from "" to ""

**Comment by mnot** [comment]
Trac update at 20080403T01:04:56: mnot@pobox.com changed type from "design" to "editorial"

**Comment by mnot** [comment]
Trac update at 20080403T13:29:45: 
* julian.reschke@gmx.de commented:

> Fixed in [238]:
> 
> Resolve #64: quoted-pair: disallow escaping of NUL, CR and LF (closes #64).


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "new" to "closed"
